contrib/globalsign/mgo/mgo.go

```diff
@@ -0,0 +1,335 @@
+	return err
+}
+
+// EnsureIndex invokes and traces Collection.EnsureIndex
+func (c *Collection) EnsureIndex(index mgo.Index) error {
+	span := newChildSpanFromContext(c.cfg)
+	err := c.Collection.EnsureIndex(index)
+	span.Finish(tracer.WithError(err))
+	return err
+}
+
+// DropIndex invokes and traces Collection.DropIndex
+func (c *Collection) DropIndex(key ...string) error {
+	span := newChildSpanFromContext(c.cfg)
+	err := c.Collection.DropIndex(key...)
+	span.Finish(tracer.WithError(err))
+	return err
+}
+
+// DropIndexName invokes and traces Collection.DropIndexName
+func (c *Collection) DropIndexName(name string) error {
+	span := newChildSpanFromContext(c.cfg)
+	err := c.Collection.DropIndexName(name)
+	span.Finish(tracer.WithError(err))
+	return err
+}
+
+// DropAllIndexes invokes and traces Collection.DropAllIndexes
+func (c *Collection) DropAllIndexes() error {
+	span := newChildSpanFromContext(c.cfg)
+	err := c.Collection.DropAllIndexes()
+	span.Finish(tracer.WithError(err))
+	return err
+}
+
+// Indexes invokes and traces Collection.Indexes
+func (c *Collection) Indexes() (indexes []mgo.Index, err error) {
+	span := newChildSpanFromContext(c.cfg)
+	indexes, err = c.Collection.Indexes()
+	span.Finish(tracer.WithError(err))
+	return indexes, err
+}
+
+// Insert invokes and traces Collectin.Insert
+func (c *Collection) Insert(docs ...interface{}) error {
+	span := newChildSpanFromContext(c.cfg)
+	err := c.Collection.Insert(docs...)
+	span.Finish(tracer.WithError(err))
+	return err
+}
+
+// Find invokes and traces Collection.Find
+func (c *Collection) Find(query interface{}) *Query {
+	return &Query{
+		Query: c.Collection.Find(query),
+		cfg:   c.cfg}
+}
+
+// FindId invokes and traces Collection.FindId
+func (c *Collection) FindId(id interface{}) *Query { // nolint
+	return &Query{
+		Query: c.Collection.FindId(id),
+		cfg:   c.cfg}
+}
+
+// Count invokes and traces Collection.Count
+func (c *Collection) Count() (n int, err error) {
+	span := newChildSpanFromContext(c.cfg)
+	n, err = c.Collection.Count()
+	span.Finish(tracer.WithError(err))
+	return n, err
+}
+
+// Bulk creates a trace ready wrapper around Collection.Bulk
+func (c *Collection) Bulk() *Bulk {
+	return &Bulk{
+		Bulk: c.Collection.Bulk(),
+		cfg:  c.cfg,
+	}
+}
+
+// Update invokes and traces Collection.Update
+func (c *Collection) Update(selector interface{}, update interface{}) error {
+	span := newChildSpanFromContext(c.cfg)
+	err := c.Collection.Update(selector, update)
+	span.Finish(tracer.WithError(err))
+	return err
+}
+
+// UpdateId invokes and traces Collection.UpdateId
+func (c *Collection) UpdateId(id interface{}, update interface{}) error { // nolint
+	span := newChildSpanFromContext(c.cfg)
+	err := c.Collection.UpdateId(id, update)
+	span.Finish(tracer.WithError(err))
+	return err
+}
+
+// UpdateAll invokes and traces Collection.UpdateAll
+func (c *Collection) UpdateAll(selector interface{}, update interface{}) (info *mgo.ChangeInfo, err error) {
```

Comment:
One thing to make sure is to check that for the latest major version of `mgo` (the one we are integrating with and promising support for) contains all these methods in all its minor versions. I think it's safe to check simply the oldest minor of the latest major.

Might not be relevant here, but it happened in the past that users could not compile their programs because of libraries making breaking changes without bumping their major version. If you find any methods missing, it might be best to remove them.